base/libgit2/types.jl

```diff
@@ -625,25 +625,24 @@ for (typ, ref, sup, fnc) in (
             (:GitTag,        :Void, :GitObject, nothing)
         )
 
-    @eval begin
-        type $typ <: $sup
-            ptr::Ptr{$ref}
-            function $typ(ptr::Ptr{$ref})
-                @assert ptr != C_NULL
-                obj = new(ptr)
-                return obj
-            end
+    @eval type $typ <: $sup
+        ptr::Ptr{$ref}
+        function $typ(ptr::Ptr{$ref})
+            @assert ptr != C_NULL
+            obj = new(ptr)
+            return obj
         end
-        if $fnc !== nothing
-            function Base.finalize(obj::$typ)
-                if obj.ptr != C_NULL
-                    ccall(($fnc, :libgit2), Void, (Ptr{$ref},), obj.ptr)
-                    obj.ptr = C_NULL
-                end
+    end
+
+    if fnc !== nothing
+        @eval function Base.finalize(obj::$typ)
```

Comment:
`finalize` has a well defined meaning of calling the finalizer and remove that from the finalizer list. This overload doesn't do that.